counsel.el

```diff
@@ -5138,6 +5141,37 @@ The optional `BUILD-ROOT' is useful for other helpers that have found
      local-files)
     compiles))
 
+(defun counsel--find-build-subdir (root)
+  "Return builds sub-directory of `ROOT', if one exists."
+  (cl-some
+   #'(lambda (x)
+       (let ((check (format "%s%s" root x)))
+         (when (file-directory-p check)
+           check)))
+   (list "build" "builds" "bld")))
+
+(defun counsel--get-build-subdirs (build-dirs)
+  "Return all subdirs of `BUILD-DIRS' sorted by access time."
+  (mapcar
+   #'car
+   (sort
+    (directory-files-and-attributes build-dirs t (rx (not (in "." ".."))))
+    #'(lambda (x y)
+        (time-less-p (nth 6 y) (nth 6 x))))))
+
+(defun counsel-compile-get-build-subdir-builds ()
```

Comment:
The name is misleading.  It essentially does what `counsel-compile-get-make-invocaton` but for sub-directories.